### PR TITLE
Update to Fastify v.3 and lc39 v.4

### PR DESCRIPTION
**Pull request by Giogia**
Hi, I updated the custom-plugin-lib to support Fastify v3 and the latest [lc39](https://github.com/mia-platform/lc39) release version.

I followed the [migration guide](https://github.com/fastify/fastify/blob/main/docs/Migration-Guide-V3.md) of Fastify to correct breaking changes and deprecations. 

The library has been tested with:
- existing tests to check that no regression has been introduced in the update process
- local execution of node microservices to assure no incompatibility has been introduced in the update process

#### Breaking Changes

`ajv` formats are no longer supported. Env variables validation can now be customized specifying a pattern field containing a regular expression instead.

##### Dependencies

- bump `@mia-platform/lc39` version from 3.3.0 to 4.0.0

- bump `@types/node` version from 13.9.1 to 15.12.0

- bump `fastify-formbody` version from 3.1.0 to 5.0.0

- bump `fastify-plugin` version from 1.6.1 to 3.0.0

- bump `eslint` version from 7.6.0 to 7.16.0

- bump `fastify-routes` version from 2.0.3 to 3.0.1

- bump `fastify-env` version from 1.0.1 to 2.1.0

- bump `typescript` version from 3.8.3 to 4.2.3

- bump `ajv` version from 6.12.6 to 8.6.0

##### Fastify

- exchanged **setschemaCompiler** with the new **setValidatorCompiler** function

- updated typescript types since Fastify v3 uses its own types (**FastifyRequest** and **FastifyReply**) rather than underlying http lib types

- addressed deprecation of attaching Object references to Fastify Request (prevent memory leaks as explained in the migration guide). Since the reference of the object is shared, any mutation will impact all requests, potentially creating security vulnerabilities or memory leaks. To achieve proper encapsulation across requests we configured a new value for each incoming request in the 'onRequest' hook (https://www.fastify.io/docs/latest/Decorators/) creating a **getter** function.

- converted error handler to use the **raw** reply object

- updated the **content type parser** function to match v3 definitions

- fixed typos and unused variables

##### Ajv

- exchanged hostname format with a regular expression pattern found in [ajv formats](https://github.com/ajv-validator/ajv-formats/blob/master/src/formats.ts). 

#### Notes

- Before updating a service that adopts the custom-plugin-lib it might be advisable to check that its JSON-schema is compatible with the Fastify latest version (`$ref` support). For more details see [here](https://github.com/fastify/fastify/blob/main/docs/Migration-Guide-V3.md#changed-schema-substitution-2023)

#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `npm test`
- [x] tests are included
- [ ] the documentation is updated or intregrated accordingly with your changes
- [x] the code will follow the lint rules and style of the project
- [x] you are not committing extraneous files or sensible data

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,13 +6,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unrealeased]
 
+### BREAKING CHANGES
+
+- [RJMR-24](https://makeitapp.atlassian.net/browse/RJMR-24) ajv formats are no longer supported. Validation can now be achieved using a pattern field and specifying a regular expression
```
Comment (fredmaggiowski) [inline CHANGELOG.md:11]: There's no need to link private tasks in the public changelogs

```suggestion
- ajv formats are no longer supported. Validation can now be achieved using a pattern field and specifying a regular expression 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,13 +6,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- [RJMR-24](https://makeitapp.atlassian.net/browse/RJMR-24) ajv formats are no longer supported. Validation can now be achieved using a pattern field and specifying a regular expression 
+
 ### Fixed
 
 - replaced expired test certificates with new ones that will expire on 3 September 3061
 ### Changed
 
 - the `getDirectServiceProxy` function now can receive a complete url as `serviceName` parameter
 
+- [RJMR-24](https://makeitapp.atlassian.net/browse/RJMR-24) update dependencies to fastify v.3 and lc39 to v4.0.0
```
Comment (fredmaggiowski) [inline CHANGELOG.md:20]:
```suggestion
- update dependencies to fastify v3 and lc39 to v4.0.0
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,13 +6,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unrealeased]
 
+### BREAKING CHANGES
+
```
Comment (dibericky) [inline CHANGELOG.md:10]: We have noticed that the update of fastify-env may have broken the support to both `required` and `default` defined at the same time in the env jsonSchema

Comment (fredmaggiowski) [reply CHANGELOG.md:10]: let's add this detail to the changelog, also let's add the ajv changelog link so the user can see the breaking changes.

**Review by fredmaggiowski — APPROVED** [review]

**Pull request merged**